Drivers/IWDG_driver/Inc/IndependentWatchdog.h

```diff
@@ -1,29 +1,22 @@
-/*
- * IndependentWatchdog.h
- *
- *  Created on: Jun. 22, 2023
- *      Author: pratyushmakkar
- */
-
 #ifndef SRC_DRIVERS_INDEPENDENTWATCHDOG_H_
 #define SRC_DRIVERS_INDEPENDENTWATCHDOG_H_
 
-
 #include <stdint.h>
 #include "main.h"
 
 class IWDG_t {
 	private:
+		IWDG_HandleTypeDef* watchdog;
 		uint8_t iwdg_prescaler;
 		uint16_t iwdg_reload;
 
 	public:
 		IWDG_t(IWDG_HandleTypeDef* watchdog);
-		IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload);
-		HAL_StatusTypeDef InitWatchdog(IWDG_HandleTypeDef* watchdog);
-		static HAL_StatusTypeDef RefreshWatchdog(IWDG_HandleTypeDef* watchdog);
+		void SetIWDGWatchdog(IWDG_HandleTypeDef* watchdog);
+		IWDG_HandleTypeDef* GetIWDGWatchdog(IWDG_HandleTypeDef* watchdog);
+		bool SetWatchdogParameters(uint8_t prescaler, uint16_t reload);
```

Comment:
I think this should be something like `UpdateWatchdogParameters`, since it updates them. Currently it implies that it sets values that haven't been otherwise set.

Reply:
Okay, after further review, I think I understand why you created this function - to allow the user to set the parameters of the watchdog (before starting the watchdog). 
I would argue (and have been basing my reviews off this assumption) that this should configuration should be done by the user before they pass the `IWDG_HandleTypeDef` to the driver, with the Init function being called from the constructor. I believe it to be the general approach that the team has been taking with other drivers thus far. Curious what you and @SnackkOverflowError think about this approach.

Reply:
Hmmm theres a lot of ways to handle this. On one hand, the watchdog parameters can be set in our IOC (which will generate code to configure them in the watchdog handle I believe). Then the handle is passed into this driver and the parameters can be retrieved from the handle. In which case I agree that this function should be named `UpdateWatchDogParameters` .
What exactly is the usecase for resetting the watchdog parameters after the driver has been created though? Is there a scenario where we want to change the frequency that we pet it at? Seems like we should agree on the watchdog parameters once and then never touch them again? If the usecase is being able to configure them in model config, IMO these params should be parameters in the constructor and set there once, then never reset during the programs lifetime.

Reply:
I agree, I feel like updating watchdog parameters after initialization could cause some unforeseen issues later on. Though, it doesn't hurt to have the function _when/if_ need arises for that.

Reply:
This function has been removed as requested.